src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js

```diff
@@ -85,33 +85,42 @@ export default class SearchWidgetPlugin extends Plugin {
      * @private
      */
     _handleInputEvent() {
-        const value = this._inputField.value.trim();
+        const formData = new FormData(this.el);
+        let encodedUri = '?';
 
         // stop search if minimum input value length has not been reached
-        if (value.length < this.options.searchWidgetMinChars) {
+        if (formData.get('search').length < this.options.searchWidgetMinChars) {
             // further clear possibly existing search results
             this._clearSuggestResults();
             return;
         }
 
-        this._suggest(value);
+        for(let [fieldName, fieldValue] of formData) {
+            if(encodedUri.charAt(encodedUri.length - 1) !== '?') {
+                encodedUri += '&';
+            }
+
+            encodedUri += `${fieldName}=` + encodeURIComponent(fieldValue);
+        }
+
+        this._suggest(encodedUri);
 
-        this.$emitter.publish('handleInputEvent', { value });
+        this.$emitter.publish('handleInputEvent', { formData });
```

Comment:
isn't this breaking if the event is suddenly emitting other data?

Reply:
Might be.
Would you prefer another Event with the whole `FormData`  and pass for the original one only the input value?

Reply:
instead of replacing it, maybe it would be enough to add it as second parameter?

Reply:
Great Idea. I'll adjust that :)

Reply:
Yes, please add a second key to the object otherwise you would break the current public API of the search widget and therefore that would end up with the consequence that your PR will be part of the next major instead of the next minor versio.

Reply:
It's done :) Sorry for the late change